artifacts/registry.py

```diff
@@ -87,6 +150,38 @@ def RegisterDefinition(self, artifact_definition):
       if source.type_indicator == definitions.TYPE_INDICATOR_ARTIFACT_GROUP:
         self._artifact_name_references.update(source.names)
 
+  @classmethod
+  def RegisterSourceType(cls, source_type_class):
+    """Registers a source type.
+
+    The source types are identified based on their type indicator.
+
+    Args:
+      source_type_class: the source type (subclass of SourceType).
+
+    Raises:
+      KeyError: if source types is already set for the corresponding
+                type indicator.
+    """
+    if source_type_class.TYPE_INDICATOR in cls._source_type_classes:
+      raise KeyError(u'Source type already set for type: {0}.'.format(
+          source_type_class.TYPE_INDICATOR))
+
+    cls._source_type_classes[source_type_class.TYPE_INDICATOR] = (
+        source_type_class)
+
+  @classmethod
+  def RegisterSourceTypes(cls, source_type_classes):
```

Comment:
Think it would be better to get rid of this and just call
```
for source_type_class in DEFAULT_SOURCE_TYPES:
 cls.register
```

in ```__init__```. Otherwise callers need to create the object and call this method with the private class variable?